lib/gringotts/gateways/we_pay.ex

```diff
@@ -0,0 +1,426 @@
+  * WePay does not support partial refunds.
+
+  ## Example
+  ```
+  iex> {:ok, refund_result} = Gringotts.refund(Gringotts.Gateways.WePay, purchase_result.id, amount)
+  ```
+  """
+  @spec refund(Money.t(), String.t(), keyword) :: {:ok | :error, Response}
+  def refund(amount, payment_id, opts) do
+    body =
+      Poison.encode!(%{
+        checkout_id: payment_id,
+        refund_reason: opts[:refund_reason]
+      })
+
+    commit(:post, "/checkout/refund/", body, opts)
+  end
+
+  @doc """
+  Stores the payment-source data for later use.
+
+  ## Note
+
+  > Currently "Recurring Payments" are not implemented.
+
+  ## Example
+  ```
+  iex> {:ok, store_result} = Gringotts.store(Gringotts.Gateways.WePay, card, opts)
+  iex> store_result.token #card token
+  ```
+  """
+  @spec store(CreditCard.t(), keyword) :: {:ok | :error, Response}
+  def store(%CreditCard{} = card, opts) do
+    body =
+      Poison.encode!(%{
+        client_id: opts[:client_id],
+        cc_number: card.number,
+        user_name: CreditCard.full_name(card),
+        email: opts[:email],
+        cvv: card.verification_code,
+        expiration_month: card.month,
+        expiration_year: card.year,
+        original_ip: opts[:original_ip],
+        original_device: opts[:original_device],
+        reference_id: opts[:reference_id],
+        address: %{
+          address1: opts[:address].street1,
+          address2: opts[:address].street2,
+          city: opts[:address].city,
+          region: opts[:address].region,
+          country: opts[:address].country,
+          postal_code: opts[:address].postal_code
+        }
+      })
+
+    commit(:post, "/credit_card/create/", body, opts)
+  end
+
+  @doc """
+  Removes card or payment info that was previously `store/2`d
+
+  Deletes previously stored payment-source data.
+
+  ## Example
+  ```
+  iex> {:ok, store_result} = Gringotts.unstore(Gringotts.Gateways.WePay, store_result.token, opts)
+  ```
+  """
+  @spec unstore(String.t(), keyword) :: {:ok | :error, Response}
+  def unstore(registration_id, opts) do
+    body =
+      Poison.encode!(%{
+        client_id: opts[:client_id],
+        client_secret: opts[:client_secret],
+        credit_card_id: registration_id
+      })
+
+    commit(:post, "/credit_card/delete/", body, opts)
+  end
+
+  ###############################################################################
+  #                                PRIVATE METHODS                              #
+  ###############################################################################
+
+  # Makes the request to wepay's network.
+  # For consistency with other gateway implementations, make your (final)
+  # network request in here, and parse it using another private method called
+  # `respond`.
+  @spec commit(atom, String.t(), String.t(), keyword) :: {:ok | :error, Response}
+  defp commit(:post, endpoint, body, opts) do
+    url = @test_url <> "#{endpoint}"
+
+    headers = [
```

Comment:
Could you also include the IP risk header?
https://developer.wepay.com/api/reference/risk_headers